src/engine/image.go

```diff
@@ -274,7 +274,7 @@ func (ir *ImageRenderer) lenWithoutANSI(text string) int {
 		text = strings.ReplaceAll(text, match[str], "")
 	}
 	stripped := regex.ReplaceAllString(color.AnsiRegex, text, "")
-	length := utf8.RuneCountInString(stripped)
+	length := runewidth.StringWidth(stripped)
 	for _, rune := range stripped {
 		length += ir.runeAdditionalWidth(rune)
```

Comment:
Doesn't this become obsolete in that case?

Reply:
Intuition tells me so, but...

If we keep that logic, there is no difference between the new output image and that of the previous version.

![normal](https://user-images.githubusercontent.com/83903009/177046657-9aee82b4-f887-4870-b026-daad7bcc7001.png)

However, if I remove that, it turns out to be something like the following:

![abnormal](https://user-images.githubusercontent.com/83903009/177046651-ac104571-87e9-4f8a-b7da-c64d07df6a36.png)

(Tested using the theme `atomic.omp.json` with `rprompt-offset` as `10`.)

No clue yet. 😅

Reply:
@lewis-yeung this part is magical at times. A real struggle. So we keep this as is and merge?

Reply:
@JanDeDobbeleer Magical indeed. Also, I think I have understood a little bit.

https://github.com/JanDeDobbeleer/oh-my-posh/blob/9f9b91e56688ed10014666a4ca815a50e794bfb9/src/engine/image.go#L420-L425

Actually, most of Nerd Font icon characters should be measured as 2-width glyphs, while on a string including these characters, `runewidth.StringWidth` returns a width just the same as `utf8.RuneCountInString` does. Thus we need to keep that logic.

I'll revert Line 277 and then it is probably ready to be merged.